### ISSUE TITLE
Some accessibility issues in GUIs

### ISSUE THREAD

**Issue by Kazaf123** [juha-h/baresip-studio#294]
**Description**
This is a bug report about components that have accessibility issues within the GUI of such an App.
Also, as the following Figure, this report focuses on the content GUI page in the App.
<img width="360" alt="MuMu20221127162448" src="https://user-images.githubusercontent.com/44084361/204125846-6f685a39-d2ee-4f65-bf13-b618c1f55454.png">
**Content**
After I applied my proposed tool to detect the content page, I discovered the following components which have accessibility issues, and I sincerely struggled you could repair these issues so that visually impaired users would use this App more conveniently.
1. The content of "Callee" has low color contrast for low-vision users to find, and it has narrow intervals with "Call to..."
2. At the bottom of the GUI, these four icons have narrow intervals to click.
**Appreciate**
I sincerely hope that these inaccessible issues could be repaired for the convenience of visually impaired users.

**Comment by juha-h** [comment]
Thanks for your comments.  Regarding Callee low contrast, it is produced by standard Android hint:
```
android:hint="@string/callee"
```
Looks like it is possible to set hint color like this:
```
android:textColorHint="@color/someColor"
```
Do you have any suggestions regarding someColor?

Regarding narrow vertical intervals, all has to fit also to devices with small screens, e.g. 5.2 inches. I need to check if there is any possibility to increase the intervals.

Regarding bottom icons, I'll check if there is a way to distribute them evenly.  In case of voicemail, there is also fifth icon than needs to fit in.

**Comment by Kazaf123** [comment]
I suggest that if the color is brown，it would be better.



---- Replied Message ----
| From | ***@***.***> |
| Date | 11/28/2022 18:49 |
| To | ***@***.***> |
| Cc | Mengxi ***@***.***>***@***.***> |
| Subject | Re: [juha-h/baresip-studio] Some accessibility issues in GUIs (Issue #294) |

Thanks for your comments. Regarding Callee low contrast, it is produced by standard Android hint:

***@***.***/callee"


Looks like it is possible to set hint color like this:

***@***.***/someColor"


Do you have any suggestions regarding someColor?

Regarding narrow vertical intervals, all has to fit also to devices with small screens, e.g. 5.2 inches. I need to check if there is any possibility to increase the intervals.

Regarding bottom icons, I'll check if there is a way to distribute them evenly. In case of voicemail, there is also fifth icon than needs to fit in.

**Comment by juha-h** [comment]
I would not like to introduce new colors to the app.  How about the same dark green color that is used in the icons at the bottom of the screen?

By the way,  I tested accessibility of MainActivity using Android Studio tool, and it didn't find anything to complain about.

**Comment by Kazaf123** [comment]
I agree with the color you give. In addition, our tool may be relatively stricter than other tools because it refers to the actual application experience of low-vision users.

**Comment by juha-h** [comment]
Commit https://github.com/juha-h/baresip-studio/commit/713736d83ab3e8524bd6f98922629c8f127110aa should solve most of the issues you reported.  It didn't increase vertical space between `Call to ...` and `Callee` fields, since the same vertical spacing is used between all lines.  Also, it should not matter, since touching `Call to ...` does nothing.

**Comment by Kazaf123** [comment]
thanks，I got it



---- Replied Message ----
| From | ***@***.***> |
| Date | 11/30/2022 12:02 |
| To | ***@***.***> |
| Cc | Mengxi ***@***.***>***@***.***> |
| Subject | Re: [juha-h/baresip-studio] Some accessibility issues in GUIs (Issue #294) |

Commit 713736d should solve most of the issues you reported. It didn't increase vertical space between Call to ... and Callee fields, since the same vertical spacing is used between all lines. Also, it should not matter, since touching Call to ... does nothing.

**Comment by juha-h** [comment]
The commit is included in version 49.2.0 issued today.
